### PR TITLE
Check for more mistakes in translatable strings

### PR DESCRIPTION
**Pull request by swissspidy**
See #61.

### CODE REVIEW

File: features/makepot.feature [hunk 1]
```diff
@@ -620,7 +617,114 @@ Feature: Generate a POT file of a WordPress project
...
+    Given an empty foo-plugin directory
+    And a foo-plugin/foo-plugin.php file:
+      """
+      <?php
+      /**
+       * Plugin Name: Plugin name
+       */
+
+      sprintf(
+        _n( '%1$s Comment (%2$s)', '%2$$s Comments (%1$s)', $number, 'foo-plugin' ),
```
Comment (schlessera) [inline features/makepot.feature]: double dollar sign here: `%2$$s`

File: src/MakePotCommand.php [hunk 2]
```diff
@@ -381,34 +427,99 @@ protected function makepot() {
...
-			WP_CLI::debug( sprintf( 'Extracted %d strings', $translations_count ), 'make-pot' );
+			if ( $translation->hasPlural() ) {
+				preg_match_all( self::SPRINTF_PLACEHOLDER_REGEX, $translation->getOriginal(), $single_placeholders );
+				$single_placeholders = $single_placeholders[0];
+
+				preg_match_all( self::SPRINTF_PLACEHOLDER_REGEX, $translation->getPlural(), $plural_placeholders );
+				$plural_placeholders = $plural_placeholders[0];
+
+				// see https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#plurals
+				if ( count( $single_placeholders ) < \count( $plural_placeholders ) ) {
```
Comment (schlessera) [inline src/MakePotCommand.php]: Two different forms of `count()` used, one relative and one absolute.

File: src/MakePotCommand.php [hunk 3]
```diff
@@ -381,34 +427,99 @@ protected function makepot() {
...
+					WP_CLI::warning( sprintf(
+						'Missing singular placeholder, needed for some languages. See https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#plurals %s',
+						$location
+					) );
+				} else {
+					sort( $single_placeholders );
```
Comment (schlessera) [inline src/MakePotCommand.php]: I'm probably missing something here, but I think it warrants a comment. The goal here is to verify the ordering of the placeholders. However, before comparing them, you sort them here. Would that put them into the same order no matter what?

Comment (swissspidy) [reply src/MakePotCommand.php]: I don't know what I was thinking here. I copied this from WPCS, but somehow messed it up. The order should be irrelevant, but different placeholders are usually not ok.

**Comment by swissspidy** [comment]
I think this one is good to go (pending passing tests and final review).

Nice side effect when this is in master:

I can use #69 to create a WordPress Trac ticket to improve all the strings in WordPress core that get flagged by this command.

**Pull request merged**